subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/ValidatePlugins.java

```diff
@@ -58,16 +63,27 @@
 @CacheableTask
 public abstract class ValidatePlugins extends DefaultTask {
 
+    @Nested
+    abstract Property<JavaLauncher> getLauncher();
+
     public ValidatePlugins() {
         getEnableStricterValidation().convention(false);
         getIgnoreFailures().convention(false);
         getFailOnWarning().convention(true);
+
+        JavaToolchainSpec toolchain = getProject().getExtensions().getByType(JavaPluginExtension.class).getToolchain();
+        JavaToolchainService service = getProject().getExtensions().getByType(JavaToolchainService.class);
+
+        getLauncher().convention(service.launcherFor(toolchain));
```

Comment:
:x: As explained it should go into the plugin instead. In that case this is `JavaGradlePluginPlugin`